app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,18 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def facebook
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for facebook is too high. [15.17/15] (http://c2.com/cgi/wiki?AbcMetric)